### PR TITLE
Refactor cbit

### PR DESCRIPTION
**Pull request by a-krawciw**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -169,13 +169,23 @@ RUN rm libtorch.zip
 ENV TORCH_LIB=/opt/torch/libtorch
 ENV LD_LIBRARY_PATH=$TORCH_LIB/lib:${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 ENV CMAKE_PREFIX_PATH=$TORCH_LIB:$CMAKE_PREFIX_PATH
+  
+RUN apt install swig liblapack-dev libmetis-dev -y -q --install-recommends
+RUN mkdir -p ${HOMEDIR}/.casadi && cd ${HOMEDIR}/.casadi \
+  && git clone https://github.com/casadi/casadi.git .
```
Comment (a-krawciw) [inline Dockerfile:175]: We may want to fork this to retain the commenting issues.

File: config/honeycomb_grizzly_default.yaml [hunk 2]
```diff
@@ -295,7 +294,7 @@
         visualize: true
     
     path_planning:
-      type: cbit
+      type: cbit # cbit for obstacle free path tracker, cbit.lidar for obstacle avoidance version
```
Comment (a-krawciw) [inline config/honeycomb_grizzly_default.yaml:297]: This comment has returned. It is irrelevant. Delete.

File: config/honeycomb_grizzly_default.yaml [hunk 3]
```diff
@@ -312,61 +311,46 @@
...
-        vel_error_weight: 1.0
-        acc_error_weight: 1.0
-        kin_error_weight: 1.0
-        lat_error_weight: 0.01
+        pose_error_cov: [1.0, 1.0, 1e5, 1e5, 1e5, 1e2]
+        vel_error_cov: [1.0, 5e2]
+        lat_error_cov: [1.0]
 
         # Misc
         command_history_length: 100
```
Comment (a-krawciw) [inline config/honeycomb_grizzly_default.yaml:356]: Update all Grizzly configs.

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp [hunk 4]
```diff
@@ -201,8 +181,10 @@ class CBIT : public BasePathPlanner {
   tactic::Timestamp prev_stamp;
 
   // Store the previously applied velocity and a sliding window history of MPC results
-  Eigen::Matrix<double, 2, 1> applied_vel;
-  std::vector<Eigen::Matrix<double, 2, 1>> vel_history;
+  Eigen::Vector2d applied_vel_;
+  Eigen::Vector2d estimated_last_vel_ = Eigen::Vector2d::Zero();
```
Comment (a-krawciw) [inline main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp:185]: Do we still need this?

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp [hunk 5]
```diff
@@ -223,5 +206,8 @@ class CBIT : public BasePathPlanner {
 
 };
 
+// Helper function for post-processing and saturating the velocity command
+Eigen::Vector2d saturateVel(const Eigen::Vector2d& applied_vel, double v_lim, double w_lim);
```
Comment (a-krawciw) [inline main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp:210]: std::clamp ?

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit_path_planner.hpp [hunk 6]
```diff
@@ -92,8 +85,10 @@ class CBITPlanner {
         // Costmap pointer
         std::shared_ptr<CBITCostmap> cbit_costmap_ptr;
 
-        CBITPlanner(CBITConfig conf_in, std::shared_ptr<CBITPath> path_in, vtr::path_planning::BasePathPlanner::RobotState& robot_state, std::shared_ptr<std::vector<Pose>> path_ptr, std::shared_ptr<CBITCostmap> costmap_ptr, std::shared_ptr<CBITCorridor> corridor_ptr, std::shared_ptr<bool>solution_ptr, std::shared_ptr<double>width_ptr, PathDirection path_direction);
+        CBITPlanner(CBITConfig conf_in, std::shared_ptr<CBITPath> path_in, vtr::path_planning::BasePathPlanner::RobotState& robot_state, std::shared_ptr<std::vector<Pose>> path_ptr, std::shared_ptr<CBITCostmap> costmap_ptr, std::shared_ptr<CBITCorridor> corridor_ptr, std::shared_ptr<bool>solution_ptr, std::shared_ptr<double>width_ptr);
```
Comment (a-krawciw) [inline main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit_path_planner.hpp:88]: Do we really need such a massive constructor?

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/utils.hpp [hunk 7]
```diff
@@ -26,12 +26,40 @@
...
+    ChainInfo getChainInfo(const tactic::LocalizationChain& robot_state);
+    
+} // namespace path_planning
+} // namespace vtr
+
+
+
+
 // Some useful Classes
 class Pose {
```
Comment (a-krawciw) [inline main/src/vtr_path_planning/include/vtr_path_planning/cbit/utils.hpp:58]: Move these out of the global namespace!

File: main/src/vtr_path_planning/src/mpc/speed_scheduler.cpp [hunk 8]
```diff
@@ -22,8 +22,7 @@
 #include <iostream>
 
 
-double ScheduleSpeed(const std::vector<double>& disc_path_curvature_xy, const std::vector<double>& disc_path_curvature_xz_yz, double VF, unsigned curr_sid, double planar_curv_weight, double profile_curv_weight, double eop_weight, double horizon_step_size, double min_vel)
-{
+double ScheduleSpeed(const std::vector<double>& disc_path_curvature_xy, const std::vector<double>& disc_path_curvature_xz_yz, double VF, unsigned curr_sid, double planar_curv_weight, double profile_curv_weight, double eop_weight, int horizon_steps, double min_vel) {
```
Comment (a-krawciw) [inline main/src/vtr_path_planning/src/mpc/speed_scheduler.cpp:25]: Can we use the CurvatureInfo approach to do curvature?

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/generate_pq.hpp
Comment (a-krawciw) [inline main/src/vtr_path_planning/include/vtr_path_planning/cbit/generate_pq.hpp:1]: I would love to get rid of this completely. We can calculate the p values in LocalizationChain now

**Review by a-krawciw — COMMENTED** [review]
I will probably leave the refactor of CBIT itself to a future PR. But I think we could still remove a lot of duplication.

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp (hunk 4 above)
Comment (a-krawciw) [reply main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp:185]: Removed

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp (hunk 5 above)
Comment (a-krawciw) [reply main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit.hpp:210]: Changed

File: main/src/vtr_path_planning/src/mpc/speed_scheduler.cpp (hunk 8 above)
Comment (a-krawciw) [reply main/src/vtr_path_planning/src/mpc/speed_scheduler.cpp]: Changed in most recent commit.

File: config/honeycomb_grizzly_default.yaml (hunk 2 above)
Comment (a-krawciw) [reply config/honeycomb_grizzly_default.yaml]: Removed

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/utils.hpp (hunk 7 above)
Comment (a-krawciw) [reply main/src/vtr_path_planning/include/vtr_path_planning/cbit/utils.hpp:55]: Leaving to a future cbit_ath_planner.cpp cleanup

File: main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit_path_planner.hpp (hunk 6 above)
Comment (a-krawciw) [reply main/src/vtr_path_planning/include/vtr_path_planning/cbit/cbit_path_planner.hpp:88]: Leaving for a future clean-up

**Review by James-R-Han — APPROVED** [review]
The core MPC is working as intended in both simulations and the real world for the jackal and warthog. 

Future work includes: resolving segfault error in cbit and automatically exporting the unicycle solver.

**Pull request merged**